unified-runtime/source/adapters/level_zero/image_common.cpp

```diff
@@ -0,0 +1,1356 @@
+//===--------- image.cpp - Level Zero Adapter -----------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <loader/ze_loader.h>
+
+#include "common.hpp"
+#ifdef UR_ADAPTER_LEVEL_ZERO_V2
```

Comment:
v1 and v2 contexts might contain different things. Let's leave it for now, but we might need to do some dependency injection or forward declaration on the context...
I'm worried this will be frustrating to edit later when people try to rely on fields from v1 or v2 context.

Reply:
Same. I am worried about this sharing of code between the V1 and V2 L0 adaptors. Only one is running at a time and so if something needs to be edited here to fix a bug in the default L0 adaptor it could break things in the other L0 adaptor as it would be easy to not check the change there.

Reply:
Yes, I agree, this is not ideal. In every place that we use either v1 or v2 context, we only rely on a few context methods that are implemented in both v1 and v2 (like getZeHandle() and getDevices()), so we might be able to address this by implementing a simple wrapper.